lib/config/flat-config-array.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * @fileoverview Flat Config Array
+ * @author Nicholas C. Zakas
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const { ConfigArray, ConfigArraySymbol } = require("@humanwhocodes/config-array");
+const { flatConfigSchema } = require("./flat-config-schema");
+const { RuleValidator } = require("./rule-validator");
+const { defaultConfig } = require("./default-config");
+const recommendedConfig = require("../../conf/eslint-recommended");
+const allConfig = require("../../conf/eslint-all");
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+const ruleValidator = new RuleValidator();
+
+/**
+ * Splits a plugin identifier in the form a/b/c into two parts: a/b and c.
+ * @param {string} identifier The identifier to parse.
+ * @returns {{objectName: string, pluginName: string}} The parts of the plugin
+ *      name.
+ */
+function splitPluginIdentifier(identifier) {
+    const parts = identifier.split("/");
+
+    return {
+        objectName: parts.pop(),
+        pluginName: parts.join("/")
+    };
+}
```

Comment:
The same logic is used in rule validator:

https://github.com/eslint/eslint/blob/86b7bddb421031db1a5c517a6ed6d1f23b388ba7/lib/config/rule-validator.js#L24-L35

so it would look for `es-syntax` rule in `node/no-unsupported-features` plugin, instead of `no-unsupported-features/es-syntax` rule in `node` plugin.

Reply:
So far this seems to be working okay with my tests. But again, this is just the first draft, so we have plenty of time to find and address errors. It will be easier to find once I start integrating.